crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -100,16 +104,25 @@ impl Persistence {
         let db = self.postgres.clone();
         tokio::spawn(
             async move {
-                let start = Instant::now();
-                let events_count = order_uids.len();
-                match boundary::store_order_events(&db, order_uids, label, Utc::now()).await {
-                    Ok(_) => {
-                        tracing::debug!(elapsed=?start.elapsed(), ?events_count, "stored order events");
-                    }
-                    Err(err) => {
-                        tracing::warn!(?err, "failed to insert order events");
-                    }
-                }
+
+                let store_order_events_inner = async {
+                    let start = Instant::now();
+                    let events_count = order_uids.len();
+                    let mut ex = db.pool.begin().await.context("begin transaction")?;
+                    store_order_events(&mut ex, order_uids, label, Utc::now()).await?;
+                    ex.commit().await?;
+                    Ok::<(Instant, usize), anyhow::Error>((start, events_count))
+                };
+
+        match store_order_events_inner.await {
+            Ok((start, events_count)) => {
+                tracing::debug!(elapsed=?start.elapsed(), ?events_count, "stored order events");
```

Comment:
Since the order events labels are not critical I think we could also make `store_order_events` handle and log the error internally to make the call sites nicer.

Reply:
@MartinquaXD I can do that, but then the behavior won't be the same as it used to, are you OK with that?

Reply:
I think this is fine. In the future please feel free to suggest these slight changes to the behavior if it leads to more readable code. If you are not sure if a change is okay just point it out and it can be discussed in the PR.